src/rendering/mod.rs

```diff
@@ -195,53 +201,60 @@ impl GlState {
         mesh: &Mesh,
         size: Size<i32, Physical>,
         scale: f64,
+        damage: &[Rectangle<i32, Logical>],
     ) -> Result<(), Gles2Error> {
-        let texture = match mesh.texture_id {
-            TextureId::Egui => self.egui_texture,
-            TextureId::User(_) => unimplemented!(),
-        };
-        gl.BindTexture(ffi::TEXTURE_2D, texture);
-
-        gl.BindBuffer(ffi::ARRAY_BUFFER, self.vertex_buffer);
-        gl.BufferData(
-            ffi::ARRAY_BUFFER,
-            (mesh.vertices.len() * std::mem::size_of::<Vertex>()) as isize,
-            mesh.vertices.as_ptr() as *const _,
-            ffi::STREAM_DRAW,
-        );
-
-        gl.BindBuffer(ffi::ELEMENT_ARRAY_BUFFER, self.element_array_buffer);
-        gl.BufferData(
-            ffi::ELEMENT_ARRAY_BUFFER,
-            (mesh.indices.len() * std::mem::size_of::<u32>()) as isize,
-            mesh.indices.as_ptr() as *const _,
-            ffi::STREAM_DRAW,
-        );
-
-        let screen_space = Rectangle::from_loc_and_size((0, 0), size);
         let clip_rect = Rectangle::from_extemities(
             (clip_rect.min.x, clip_rect.min.y),
             (clip_rect.max.x, clip_rect.max.y),
-        );
-        let scissor_box = clip_rect
-            .to_f64()
-            .to_physical(scale)
-            .intersection(screen_space.to_f64())
-            .unwrap()
-            .to_i32_round();
-
-        gl.Scissor(
-            scissor_box.loc.x,
-            scissor_box.loc.y,
-            scissor_box.size.w,
-            scissor_box.size.h,
-        );
-        gl.DrawElements(
-            ffi::TRIANGLES,
-            mesh.indices.len() as i32,
-            ffi::UNSIGNED_INT,
-            std::ptr::null(),
-        );
+        )
+        .to_f64();
+
+        if let Some(damage) = damage.iter().find(|d| d.overlaps(clip_rect.to_i32_round())) {
```

Comment:
There may potentially be multiple damage-rectangles, that overlap with out clip-rectangle.
So this should really be a for-loop and `filter` instead of `find`.

Reply:
Ah, sure! That make sense